Automattic/wp-calypso
# Move details about refund policies for cart items to `CheckoutTerms`

#### Proposed Changes

🚨 This description has been updated after several rounds of reviews and fixes.

A while back, in #63489, we added more detailed info to the checkout page about refund windows for different types of cart items. That change sparked a discussion about whether this was really the best way to display those details and we [concluded](https://github.com/Automattic/wp-calypso/pull/63489#issuecomment-1144957120) that no, `CheckoutTerms` (i.e. just above the `Pay now` button) would be a better placement.

This PR reduces the information in `WPCheckoutOrderSummary` (i.e. the right sidebar) to a single row and adds a new `RefundPolicies` component that replaces `DomainRefundPolicy` in the checkout terms. `RefundPolicies` displays information about refund windows for all refundable product types.

I have also incorporated a request from Nomado to change how domain refunds are communicated in the checkout summary. The changes here will probably be followed up by another PR that adds some more text to the checkout summary for cases when all checkout items are domains.

In `WPCheckoutOrderSummary`, the following logic applies:

1. If the cart contains no refundable products, display nothing
2. If the cart contains only domain registrations/renewals, display nothing
3. If the cart contains any non-refundable products, display nothing
4. If there is a single refund window in the cart AND there is a plan+free domain bundle, display `%(days)d-day money back guarantee for %(planName)s`
5. If there is a single refund window in the cart and no plan+domain bundle, display `%(days)d-day money back guarantee`
6. If there is a monthly plan+paid domain "bundle" in the cart, display `%(days)d-day money back guarantee for %(planName)s`
7. If the cart contains only a plan renewal and a domain renewal, display `%(days)d-day money back guarantee for %(planName)s`
8. If there are multiple refund windows in the cart, display `%(minimumRefundWindow)d-day full money back guarantee`

| Products | Screenshot |
| - | - |
| Yearly plan, bundled domain | ![01-yealy-plan-domain-bundle](https://user-images.githubusercontent.com/1101677/194013969-2c94be3d-aa89-4c43-8489-7614cd4a79e6.png) |
| Yearly plan, bundled domain, Professional Email trial | ![02-yearly-plan-domain-bundle-professional-email](https://user-images.githubusercontent.com/1101677/194013977-f61d627b-6e67-47c9-b641-98be1a16126e.png) |
| Yearly plan, bundled domain, add-on | ![03-yearly-plan-domain-bundle-add-on](https://user-images.githubusercontent.com/1101677/194013980-2858aac5-e6ed-4014-8d0d-fb7a622117e4.png) |
| Monthly plan, "bundled" domain, add-on | ![04-monthly-plan-domain-bundle-add-on](https://user-images.githubusercontent.com/1101677/194013983-5c8466c2-df82-47dc-b8e5-9d7c4aa1cad9.png) |
| Monthly plan, "bundled" domain | ![05-montly-plan-domain-bundle](https://user-images.githubusercontent.com/1101677/194013987-6bf44cb8-0d90-48e8-8a62-599f4d872717.png) |
| DIFM Lite | ![difm-after](https://user-images.githubusercontent.com/1101677/191489013-84987807-ddc6-42bc-8839-2bfc1242b6ba.png) |


#### Testing Instructions



We want to test with a few different types of products, so bear with me.

1. Go to `/start/onboarding-with-email`
2. Add a domain, a mailbox and a plan and proceed to checkout
3. Ensure that a `14-day money back guarantee for {{ planName }}` row is visible in the right sidebar
4. Ensure that the following text row is visible in `CheckoutTerms`: `You understand that domain name refunds are limited to 96 hours after registration and yearly plan refunds are limited to 14 days after purchase. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.`
5. Complete checkout
6. Navigate to `/setup?siteSlug=<site_slug>`
7. Check the `Hire a professional to design my website` checkbox and proceed all the way to checkout
8. Ensure that **no** `Money back guarantee` row is visible in the right sidebar
9. Ensure that **no** information about refunds is visible in `CheckoutTerms`
10. Empty the cart
11. Set up the mailbox you bought earlier and then click `Renew now` on the email management page
12. Ensure that a `14-day money back guarantee` row is visible in the right sidebar 
13. Ensure that the following text row is visible in `CheckoutTerms`: `You understand that refunds are limited to 14 days after purchase or renewal for non-domain products with yearly subscriptions.`
14. Empty the cart
15. Add a new domain to your cart (`/domains/add/:site_slug`) and proceed to checkout
16. Ensure that **no** `Money back guarantee` row is visible in the right sidebar
17. Ensure that **no** information about refunds is visible in `CheckoutTerms`

#### Pre-merge Checklist






Related to #
